packages/tracing/tests/e2e/tracer.test.ts

```diff
@@ -148,13 +148,28 @@ describe('Tracer integration tests', () => {
         // Assert that the subsegment name is the expected one
         expect(handlerSubsegment.name).toBe('## index.handler');
         if (handlerSubsegment?.subsegments !== undefined) {
-          // Assert that there're two subsegments
-          expect(handlerSubsegment?.subsegments?.length).toBe(2);
+          // Assert that there are three subsegments
+          expect(handlerSubsegment?.subsegments?.length).toBe(3);
 
-          const [ AWSSDKSubsegment1, AWSSDKSubsegment2 ] = handlerSubsegment?.subsegments;
-          // Assert that the subsegment names is the expected ones
-          expect(AWSSDKSubsegment1.name).toBe('DynamoDB');
-          expect(AWSSDKSubsegment2.name).toBe('DynamoDB');
+          // Sort the subsegments by name
+          const dynamoDBSubsegments: ParsedDocument[] = [];
+          const httpSubsegment: ParsedDocument[] = [];
+          const otherSegments: ParsedDocument[] = [];
+          handlerSubsegment?.subsegments.forEach(subsegment => {
+            if (subsegment.name === 'DynamoDB') {
+              dynamoDBSubsegments.push(subsegment);
+            } else if (subsegment.name === 'httpbin.org') {
+              httpSubsegment.push(subsegment);
+            } else {
+              otherSegments.push(subsegment);
+            }
+          });
```

Comment:
1. I don't think that this is sorting. More like splitting them?
2. This duplicates with many other tests. Should we extract it into a function like this?

```typescript
const {dynamoDBSubsegments, httpSubsegment, otherSegments} = splitSegmentByName(handlerSubsegment?.subsegments);
```

Reply:
I see that you may also have other types.  The more generic may be:

```typescript
const expectedSubsegmentNames = ['DynamoDB', 'httpbin.org']; 
const subsegmentsByName = splitSegmentByName(
  handlerSubsegment?.subsegments, 
  expectedSubsegmentNames
);

expect(subsegmentsByName['DynamoDB'].length).toBe(2);
expect(subsegmentsByName['httpbin.org'].length).toBe(2);
expect(subsegmentsByName['others'].length).toBe(2);
```